compiler-plugin/src/main/kotlin/com/google/devtools/ksp/processing/impl/ResolverImpl.kt

```diff
@@ -691,7 +691,8 @@ class ResolverImpl(
                 builtIns.arrayType.replace(typeArgs)
             }
             else -> {
-                getClassDeclarationByName(psiType.canonicalText)?.asStarProjectedType() ?: KSErrorType
+                getClassDeclarationByName(psiType.canonicalText)?.asStarProjectedType()
+                    ?: KSErrorType(psiType.canonicalText)
```

Comment:
Any specific reason to use canonical text? I think you are using simple names for implementations in analysis api?

Reply:
Yes, the idea is to try including as much _type_ information as available. If additional qualifiers are resolved by the compiler - include them. I guess you're right about the things I did in AA, so I went and tried to use qualified names there as well. See `KtType.render() -> KtClassErrorType.qualifiers`